torch/testing/_internal/common_device_type.py

```diff
@@ -329,13 +326,17 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
     def run(self, result=None):
         super().run(result=result)
         # Early terminate test if _stop_test_suite is set.
+        # Using this flag avoids unnecessary check invoke of torch.cuda.synchronize()
```

Comment:
I don't think I understand this comment, sorry

Reply:
so.. my plan is to 
1. keep `DeviceGenericTestBase` class to use `self._stop_test_suite` variable - which is only set when RTE occurs on test cases. 
2. while the common_utils `TestCase` class to use `self. _should_stop_test_suite()` function, and this function calls `torch.cuda.synchronize()` all the time regardless of whether it is a CPU or GPU test, since it can't tell.